prive/datasets/dataset.py

```diff
@@ -63,76 +65,210 @@ def replace(self, record_in, record_out):
+    and the data description is a dictionary.
+    """
 
-    """Base class for data objects """
-    def __init__(self, input_path):
-        self.dataset, self.description = self.read(input_path)
+    def __init__(self, dataset, description):
+        super().__init__(dataset, description)
 
-    def read(self, input_path):
+    @classmethod
+    def read(cls, filepath):
         """
-        Read csv and json files dataframe and and dictionary respectively.
+        Read csv and json files for dataframe and description dictionary respectively.
+
+        Parameters
+        ----------
+        filepath (str): Path where the csv and json file are located
+
+        Returns
+        -------
+        A TabularDataset instantiated object.
 
-        :param input_path: A string with the path where the csv and json file are located
-        :return: A pandas dataframe and a dictionary with data description.
         """
-        with open(f'{input_path}.json') as f:
+        with open(f'{filepath}.json') as f:
             description = json.load(f)
 
-        #TODO: something like this to determine types and columns to be read in the dataframe
-        #dtypes = {cd['name']: _get_dtype(cd) for cd in self.description['columns']}
-        #columns = self.description['columns']
+        # TODO: something like this to determine types
+        #  and columns to be read in the dataframe once we ha a defined json
+        # dtypes = {cd['name']: _get_dtype(cd) for cd in self.description['columns']}
+        # columns = self.description['columns']
 
-        dataset = pd.read_csv(f'{input_path}.csv')
+        dataset = pd.read_csv(f'{filepath}.csv')
 
-        return dataset, description
+        return cls(dataset, description)
 
-    def write(self, output_path):
+    def write(self, filepath):
         """
         Write dataset and description to file
+
+        Parameters
+        ----------
+        filepath (str): Path where the csv and json file are saved
         """
-        pass
+
+        with open(f'{filepath}.csv', 'w') as fp:
+            json.dump(dict, fp)
+
+        self.dataset.to_csv(filepath)
 
     def sample(self, n_samples):
         """
-        Sample from dataset a set of records.
-        """
-        pass
+        Sample from a TabularDataset object a set of records.
+
+        Parameters
+        ----------
+        n_samples (int): Number of records to sample
+
+        Returns
+        -------
+
+        A TabularDataset object with a sample of the records of the original object.
+
 
-    def get_record(self, record_ids):
         """
-        Select and return a record(s).
+        return TabularDataset(dataset=self.dataset.sample(n_samples), description=self.description)
+
+    def get_records(self, record_ids):
         """
-        pass
+        Get a record from the TabularDataset object
+
+        Parameters
+        ----------
+        record_ids (List[int]): List of indexes of records to retrieve
```

Comment:
Type hints in 3.9 now support built-in types, so you can now write list[int], we should decide whether we want to follow this in our docstrings as well (as opposed to capitalisation).

Reply:
changes to non capitalisation